docs/content/learn/tutorials/component-definition/_index.md

```diff
@@ -0,0 +1,680 @@
+{{% /tab %}}
+{{% tab %}}
+{{< highlight yaml "linenos=table" >}}
+---
+component-definition:
+  ...
+  components:
+    - uuid: 91f646c5-b1b6-4786-9ec3-2305a044e217
+      type: software
+      title: MongoDB
+      description: >-
+        MongoDB is a source-available cross-platform document-oriented database
+        program. Classified as a NoSQL database program, MongoDB uses JSON-like
+        documents with optional schemas.
+      purpose: Backend NoSQL database
+      responsible-roles:
+        - role-id: supplier
+          party-uuid: ef7c799a-c50e-49ab-83e0-515e989e6df1
+      protocols:
+        - name: Transmission Control
+      control-implementations:
+        - Implementation details...
+
+{{< /highlight >}}
+
+A component is represented in the `components` object array shown in line 4.   Each component requires a unique identifier, which is provided using the `uuid` property.
+
+The `type` item is very useful in classifying components.  Some common examples include interconnection, hardware, software, service, policy, process / procedure,  plan, guidance, and standard. In our example, we’ll set the MongoDB component to a *“software"* type.  
+Next, the component needs to have title and description.
+The optional `responsible-roles` item can be used to reference one or more roles with responsibility for performing a function relative to the component.  In this tutorial, will reference the supplier.
+{{% /tab %}}
+{{% /tabs %}}
+
+### Protocol
+
+Another optional but valuable sub-element of `<component>` is [`<protocol>`](/reference/latest/component-definition/xml-reference/#/component-definition/component/protocol).  Generally, `<protocol>` elements are only applicable when dealing with a `<component>` of type *"service"*.  In our example, we'll assume that installations of MongoDB will result in an installation of mongod (Mongo Daemon) which runs the MongoDB database as a process and uses the following protocols:
+
+| Port| Transport | Direction | Description |
+| ----- | --------- | --------- | ----------- |
+| 27017 | TCP | Inbound | The default port for mongod and mongos instances. |
+| 27018 | TCP | Inbound | The default port when running with --shardsvr runtime operation. |
+| 27009 | TCP | Inbound-Outbound | The default port when running with --configsvr runtime operation |
+
+This is represented below.
+
+{{< tabs XML JSON YAML >}}
+{{% tab %}}
+{{< highlight xml "linenos=table" >}}
+<component-definition xmlns="http://csrc.nist.gov/ns/oscal/1.0"
+    uuid="a7ba800c-a432-44cd-9075-0862cd66da6b">
+  ...  
+  <component uuid="3e5c92b8-870c-4313-871a-f4307bf2eaf0" type="service">
+      ...
+      <protocol name="Transmission Control">
+        <title>MongoDB requires use of TCP.</title>
+        <port-range start="27017" end="27017" transport="TCP" />
+        <port-range start="27018" end="27018" transport="TCP" />
+        <port-range start="27019" end="27019" transport="TCP" />
```

Comment:
You should demonstrate using the `name` and `title` elements within a protocol. There should be an individual protocol for each port. The names should probably be `mongod`, `shardsrv`, and `configsrv` or something similar. The title would then be a human readable name for these services.